plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/runtime/jobs/DataSourceMonitorJob.java

```diff
@@ -95,21 +95,21 @@ protected IStatus run(DBRProgressMonitor monitor) {
 
     protected void doJob() {
         final DBPWorkspace workspace = platform.getWorkspace();
-        checkDataSourceAliveInWorkspace(workspace, () -> getLastUserActivityTime(lastPingTime));
+        checkDataSourceAliveInWorkspace(workspace, getLastUserActivityTime(lastPingTime));
     }
 
-    protected void checkDataSourceAliveInWorkspace(DBPWorkspace workspace, LongSupplier supplier) {
+    protected void checkDataSourceAliveInWorkspace(DBPWorkspace workspace, long lastUserActivityTime) {
         for (DBPProject project : workspace.getProjects()) {
             if (project.isOpen() && project.isRegistryLoaded()) {
                 DBPDataSourceRegistry dataSourceRegistry = project.getDataSourceRegistry();
                 for (DBPDataSourceContainer ds : dataSourceRegistry.getDataSources()) {
-                    checkDataSourceAlive(ds, supplier, workspace.getActiveProject().getWorkspaceSession());
+                    checkDataSourceAlive(ds, lastUserActivityTime);
```

Comment:
Note for reviewers: here was an NPE caused by the return value of `workspace.getActiveProject()` being null. We can get a project from a data source container that we conveniently pass to the same function.